gcc/rust/parse/rust-parse-impl.h

```diff
@@ -6375,6 +6375,9 @@ template <typename ManagedTokenSource>
 AST::GenericArgs
 Parser<ManagedTokenSource>::parse_path_generic_args ()
 {
+  if (lexer.peek_token ()->get_id () == LEFT_SHIFT)
```

Comment:
I think we cannot simply peek a token here and expect it to be either a left shift or a left angle, this will not scale with deeply nested generics.